Compilation issues when using gcc-12

```bash
In file included from /usr/include/c++/12/bits/alloc_traits.h:33,
                 from /usr/include/c++/12/ext/alloc_traits.h:34,
                 from /usr/include/c++/12/bits/basic_string.h:39,
                 from /usr/include/c++/12/string:53,
                 from /usr/include/c++/12/stdexcept:39,
                 from /usr/include/c++/12/system_error:41,
                 from /usr/include/c++/12/mutex:41,
                 from /home/jamstutz/opt/anari/include/anari/backend/helium/LockableObject.h:7,
                 from /home/jamstutz/opt/anari/include/anari/backend/helium/BaseObject.h:6,
                 from /home/jamstutz/src/VisRTX/devices/rtx/Object.h:35,
                 from /home/jamstutz/src/VisRTX/devices/rtx/RegisteredObject.h:34,
                 from /home/jamstutz/src/VisRTX/devices/rtx/scene/surface/geometry/Geometry.h:34,
                 from /home/jamstutz/src/VisRTX/devices/rtx/scene/surface/geometry/Geometry.cpp:32:
/usr/include/c++/12/bits/stl_construct.h: In instantiation of ‘void std::_Construct(_Tp*, _Args&& ...) [with _Tp = visrtx::GeometryGPUData; _Args = {}]’:
/usr/include/c++/12/bits/stl_uninitialized.h:638:18:   required from ‘static _ForwardIterator std::__uninitialized_default_n_1<_TrivialValueType>::__uninit_default_n(_ForwardIterator, _Size) [with _ForwardIterator = visrtx::GeometryGPUData*; _Size = long unsigned int; bool _TrivialValueType = false]’
/usr/include/c++/12/bits/stl_uninitialized.h:701:20:   required from ‘_ForwardIterator std::__uninitialized_default_n(_ForwardIterator, _Size) [with _ForwardIterator = visrtx::GeometryGPUData*; _Size = long unsigned int]’
/usr/include/c++/12/bits/stl_uninitialized.h:766:44:   required from ‘_ForwardIterator std::__uninitialized_default_n_a(_ForwardIterator, _Size, allocator<_Tp>&) [with _ForwardIterator = visrtx::GeometryGPUData*; _Size = long unsigned int; _Tp = visrtx::GeometryGPUData]’
/usr/include/c++/12/bits/vector.tcc:644:35:   required from ‘void std::vector<_Tp, _Alloc>::_M_default_append(size_type) [with _Tp = visrtx::GeometryGPUData; _Alloc = std::allocator<visrtx::GeometryGPUData>; size_type = long unsigned int]’
/usr/include/c++/12/bits/stl_vector.h:1013:4:   required from ‘void std::vector<_Tp, _Alloc>::resize(size_type) [with _Tp = visrtx::GeometryGPUData; _Alloc = std::allocator<visrtx::GeometryGPUData>; size_type = long unsigned int]’
/home/jamstutz/src/VisRTX/devices/rtx/utility/HostDeviceArray.h:89:23:   required from ‘void visrtx::HostDeviceArray<T>::resize(size_t, bool) [with T = visrtx::GeometryGPUData; size_t = long unsigned int]’
/home/jamstutz/src/VisRTX/devices/rtx/utility/DeviceObjectArray.h:94:21:   required from ‘visrtx::DeviceObjectIndex visrtx::DeviceObjectArray<T>::alloc(void*) [with T = visrtx::GeometryGPUData; visrtx::DeviceObjectIndex = int]’
/home/jamstutz/src/VisRTX/devices/rtx/RegisteredObject.h:105:22:   required from ‘void visrtx::RegisteredObject<GPU_DATA_T>::setRegistry(array_t&) [with GPU_DATA_T = visrtx::GeometryGPUData; array_t = visrtx::DeviceObjectArray<visrtx::GeometryGPUData>]’
/home/jamstutz/src/VisRTX/devices/rtx/scene/surface/geometry/Geometry.cpp:79:14:   required from here
/usr/include/c++/12/bits/stl_construct.h:119:7: error: use of deleted function ‘visrtx::GeometryGPUData::GeometryGPUData()’
  119 |       ::new((void*)__p) _Tp(std::forward<_Args>(__args)...);
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/jamstutz/src/VisRTX/devices/rtx/Object.h:41:
/home/jamstutz/src/VisRTX/devices/rtx/gpu/gpu_objects.h:195:8: note: ‘visrtx::GeometryGPUData::GeometryGPUData()’ is implicitly deleted because the default definition would be ill-formed:
  195 | struct GeometryGPUData
      |        ^~~~~~~~~~~~~~~
/home/jamstutz/src/VisRTX/devices/rtx/gpu/gpu_objects.h:209:24: error: union member ‘visrtx::GeometryGPUData::<unnamed union>::neural’ with non-trivial ‘visrtx::NeuralGeometryData::NeuralGeometryData()’
  209 |     NeuralGeometryData neural;
      |                        ^~~~~~
```

--- a/devices/rtx/gpu/gpu_objects.h
+++ b/devices/rtx/gpu/gpu_objects.h
@@ -192,7 +192,8 @@ struct NeuralGeometryData
   __half *weights[NEURAL_NB_MAX_LAYERS]; // Array of weight matrices
   __half *biases[NEURAL_NB_MAX_LAYERS]; // Array of bias vectors
   uint32_t nb_layers;
-  box3 bounds;
+  vec3 boundMin;
+  vec3 boundMax;
   float threshold;
 };
 #endif

--- a/devices/rtx/scene/Intersectors_ptx.cu
+++ b/devices/rtx/scene/Intersectors_ptx.cu
@@ -367,7 +367,7 @@ VISRTX_DEVICE float __optix_enabled__forwardSDF(
 VISRTX_DEVICE void intersectNeural(const GeometryGPUData &geometryData)
 {
   const auto &neuralData = geometryData.neural;
-  const box3 bounds = neuralData.bounds;
+  const box3 bounds = { neuralData.boundMin, neuralData.boundMax };
   const vec3 &ro = ray::localOrigin();
   const vec3 &rd = ray::localDirection();
   float t0, t1;

--- a/devices/rtx/scene/surface/geometry/Neural.cpp
+++ b/devices/rtx/scene/surface/geometry/Neural.cpp
@@ -63,11 +63,8 @@ GeometryGPUData Neural::gpuData() const
   neural.nb_layers = m_layers.size() / 2;
 
   // Read bounds from parameters
-  vec3 aabb_min = getParam<glm::vec3>("aabb_min", glm::vec3(-1.f));
-  vec3 aabb_max = getParam<glm::vec3>("aabb_max", glm::vec3(1.f));
-  m_aabb.lower = aabb_min;
-  m_aabb.upper = aabb_max;
-  neural.bounds = m_aabb;
+  neural.boundMin = getParam<glm::vec3>("aabb_min", glm::vec3(-1.f));
+  neural.boundMax = getParam<glm::vec3>("aabb_max", glm::vec3(1.f));
 
   neural.threshold = getParam<float>("threshold", 0.1f);
 

